### PR TITLE
Add generateComposePreviewDesktopTests Gradle extension

### PR DESCRIPTION
**Pull request by takahirom**
# What
Add `roborazzi { generateComposePreviewDesktopTests { ... } }`: generated Compose Preview screenshot tests for the Compose Desktop (JVM) target, the desktop sibling of `generateComposePreviewRobolectricTests`.

- Hooks Kotlin JVM targets: with exactly one JVM target the tests are generated for it automatically; with multiple targets `targetName` must be set (clear error listing the targets); plain-JVM (`org.jetbrains.kotlin.jvm`) projects are supported.
- Generates JUnit4 `Parameterized` test classes that drive `DesktopComposePreviewTester` (custom testers via `testerQualifiedClassName`, sharding via `generatedTestClassCount`/`maxParallelForks`, `includePrivatePreviews`, `annotationFilter`).
- No `RoboPreviewExclude` default filter yet: roborazzi-annotations is an Android library, so the marker annotations are not usable from commonMain/desktop (planned as a follow-up).
- Warns when both preview generators are enabled in one module without `separateOutputDirs` (same file names would overwrite each other).
- Integration tests: new `sample-generate-preview-desktop-tests` template + 9 test cases (record, disable, target resolution, missing dependency, private previews, custom tester, sharding).
- Fixes a TestKit classloader split by adding `org.jetbrains.kotlin:compose-compiler-gradle-plugin` to the integration plugin classpath — without it the compose compiler silently no-ops for JVM modules built inside the composite.

# Why
Proposals/02 phase 2: the tester module from the previous PR covers manual use; this makes desktop preview tests zero-boilerplate, matching the Android experience.

Stacked on #898.

## Summary by CodeRabbit

* **New Features**
  * Added support for generating Compose Desktop preview screenshot tests for JVM and Kotlin Multiplatform projects.
  * Added configuration for target selection, package scanning, private previews, annotation filtering, test sharding, and custom preview testers.
  * Added support for loading custom desktop preview testers by class name.
  * Generated tests are automatically connected to desktop test tasks.

* **Tests**
  * Added coverage for generated screenshots, configuration validation, custom testers, private previews, and multiple JVM targets.

### CODE REVIEW

File: include-build/roborazzi-gradle-plugin/src/integrationTest/java/io/github/takahirom/roborazzi/DesktopPreviewGenerateTest.kt [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+
+  @Test
+  fun whenTargetNameSpecifiedShouldBeRecorded() {
+    DesktopPreviewModule(RoborazziGradleRootProject(testProjectDir), testProjectDir).apply {
+      buildGradle.targetName = "desktop"
+
+      record()
+
+      checkHasImages()
+    }
```
[the remark this answers is not in the document]
Comment (takahirom) [reply include-build/roborazzi-gradle-plugin/src/integrationTest/java/io/github/takahirom/roborazzi/DesktopPreviewGenerateTest.kt:53]: This scenario can no longer be built: Kotlin Gradle Plugin 2.2 hard-fails a project that declares two `jvm()` targets ("jvm() Kotlin Target Already Declared" from checkKotlinGradlePluginConfigurationErrors), so a passing multi-target + targetName test cannot exist. The multi-target *error* path is still covered by `whenMultipleJvmTargetsWithoutTargetNameShouldFail` (Roborazzi's configuration-time error fires before KGP's execution-time check), and targetName selection/validation is covered by `whenTargetNameSpecifiedShouldBeRecorded` / `whenWrongTargetNameShouldFail`.

File: include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/DesktopGeneratePreviewTestsConfigurator.kt [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+      )
+    }
+  }
+
+  val requiredLibraries = listOf(
+    "io.github.takahirom.roborazzi:roborazzi-compose-desktop-preview-scanner-support",
+    "junit:junit",
+    "io.github.sergio-sastre.ComposablePreviewScanner:android",
+  )
+  requiredLibraries.forEach { dependencies.checkExists(it) }
```
[the remark this answers is not in the document]
Comment (takahirom) [reply include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/DesktopGeneratePreviewTestsConfigurator.kt:208]: Partially addressed in #902: the direct `junit:junit` requirement was dropped (it usually arrives transitively via kotlin("test")), which removes the false-rejection case. The remaining check (scanner-support module + ComposablePreviewScanner) intentionally stays a direct-declaration scan: both must be declared directly by the consumer anyway (the support module ships the scanner as compileOnly), and resolving the test classpath at configuration time is the performance trap the Android configurator also avoids (see the BOM note there referencing gradle/gradle#2298).

File: include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/GenerateComposePreviewDesktopTestsExtension.kt [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+    val generatedClassFQDN =
+      "com.github.takahirom.roborazzi.RoborazziDesktopPreviewParameterizedTests"
+    val packageName = generatedClassFQDN.substringBeforeLast(".")
+    val baseClassName = generatedClassFQDN.substringAfterLast(".")
+    val directory = File(testDir, packageName.replace(".", "/"))
+    directory.mkdirs()
+
+    // Delete old generated test files to avoid conflicts when changing generatedTestClassCount
+    directory.listFiles()?.filter { it.extension == "kt" }?.forEach { it.delete() }
```
[the remark this answers is not in the document]
Comment (takahirom) [reply include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/GenerateComposePreviewDesktopTestsExtension.kt:131]: Fixed in #902 (abb857f3): the cleanup now only deletes files matching this task's generated class name prefix, so generators sharing a directory can't wipe each other's output. (In practice the directories are per-variant — `preview-screenshot/desktop` vs `preview-screenshot/debug` — but the scoped filter makes it safe regardless.)
